src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -26,24 +27,42 @@
     private long threadLocalSetupTimeDiff = 0;
     private long pairHMMSetupTime = 0;
 
+    public enum Implementation {AVX, OMP}
+
     private final PairHMMNativeBinding pairHmm;
 
     //Hold the mapping between haplotype and index in the list of Haplotypes passed to initialize
     //Use this mapping in computeLikelihoods to find the likelihood value corresponding to a given Haplotype
     private final Map<Haplotype, Integer> haplotypeToHaplotypeListIdxMap = new LinkedHashMap<>();
     private HaplotypeDataHolder[] mHaplotypeDataArray;
 
-    public VectorLoglessPairHMM(PairHMMNativeArguments args) throws UserException.HardwareFeatureException {
-        // TODO: connect GATK temp directory
-        final boolean isSupported = new IntelPairHmm().load(null);
-
-        if (!isSupported) {
-            throw new UserException.HardwareFeatureException("Machine does not support AVX PairHMM.");
+    public VectorLoglessPairHMM(Implementation implementation,
+                                PairHMMNativeArguments pairHmmArgs) throws UserException.HardwareFeatureException {
+        final boolean isSupported;
+
+        switch (implementation) {
+            case AVX:
+                isSupported = new IntelPairHmm().load();
```

Comment:
Nitpicky, but I would generally avoid reusing `isSupported` in two cases like this since they really have nothing to do with each other.  Could you just declare it separately in each block, maybe give them distinct names, ie isAVXSupported and isOMPSupported.